tests/reftests/filter-operators.test

```diff
@@ -0,0 +1,327 @@
+  - install ok 1 [required by or]
+  - install or 1
+### :: eq ::
+### <pkg:eq.1>
+opam-version: "2.0"
+depends: [
+  "ok"
+  "a" { 2=2 }
+  "ko-int" { 2=3 }
+  "b" { "a" = "a" }
+  "ko-string" { "a" = "b" }
+  "c" { true = true }
+  "ko-bool" { true = false }
+]
+### opam install eq
+The following actions would be performed:
+=== install 5 packages
+  - install a  1 [required by eq]
+  - install b  1 [required by eq]
+  - install c  1 [required by eq]
+  - install eq 1
+  - install ok 1 [required by eq]
+### :: neq ::
+### <pkg:neq.1>
+opam-version: "2.0"
+depends: [
+  "ok"
+  "a" { 2!=3 }
+  "ko-int" { 2!=2 }
+  "b" { "a" != "b" }
+  "ko-string" { "b" != "b" }
+  "c" { true != false }
+  "ko-bool" { true != true }
+]
+### opam install neq
+The following actions would be performed:
+=== install 5 packages
+  - install a   1 [required by neq]
+  - install b   1 [required by neq]
+  - install c   1 [required by neq]
+  - install neq 1
+  - install ok  1 [required by neq]
+### :: lt ::
+### <pkg:lt.1>
+opam-version: "2.0"
+depends: [
+  "ok"
+  "a" { 2<3 }
+  "ko-int" { 3<2 }
+  "b" { "a" < "b" }
+  "ko-string" { "b" < "a" }
+  "c" { false < true }
+  "ko-bool" { true < false }
+]
+### opam install lt
+The following actions would be performed:
+=== install 5 packages
+  - install a  1 [required by lt]
+  - install b  1 [required by lt]
+  - install c  1 [required by lt]
+  - install lt 1
+  - install ok 1 [required by lt]
+### :: lteq ::
+### <pkg:lteq.1>
+opam-version: "2.0"
+depends: [
+  "ok"
+  "a" { 2<=3 & 2<=2 }
+  "ko-int" { 3<=2 }
+  "b" { "a" <= "b" & "a" <= "a" }
+  "ko-string" { "b" <= "a" }
+  "c" { false <= true & false <= false }
+  "ko-bool" { true <= false }
+]
+### opam install lteq
+The following actions would be performed:
+=== install 5 packages
+  - install a    1 [required by lteq]
+  - install b    1 [required by lteq]
+  - install c    1 [required by lteq]
+  - install lteq 1
+  - install ok   1 [required by lteq]
+### :: gt ::
+### <pkg:gt.1>
+opam-version: "2.0"
+depends: [
+  "ok"
+  "a" { 4>3 }
+  "ko-int" { 3>4 }
+  "b" { "d" > "b" }
+  "ko-string" { "b" > "d" }
+  "c" { true > false }
+  "ko-bool" { false > true }
+]
+### opam install gt
+The following actions would be performed:
+=== install 5 packages
+  - install a  1 [required by gt]
+  - install b  1 [required by gt]
+  - install c  1 [required by gt]
+  - install gt 1
+  - install ok 1 [required by gt]
+### :: gteq ::
+### <pkg:gteq.1>
+opam-version: "2.0"
+depends: [
+  "ok"
+  "a" { 4>=3 & 4>=4 }
+  "ko-int" { 3>=4 }
+  "b" { "d" >= "b" & "d" >= "d" }
+  "ko-string" { "b" >= "d" }
+  "c" { true >= false & true >= true }
+  "ko-bool" { false >= true }
+]
+### opam install gteq
+The following actions would be performed:
+=== install 5 packages
+  - install a    1 [required by gteq]
+  - install b    1 [required by gteq]
+  - install c    1 [required by gteq]
+  - install gteq 1
+  - install ok   1 [required by gteq]
+### :: Combination ::
```

Comment:
I think it's definitely good to have some tests for this, but I'm not sure that reftests are the correct harness? Wouldn't it be better to be testing the filter parser then evaluation of filters separately (alcotest-style or some such), rather than invoking the whole of opam machinery to test this?

I'm pretty sure the tests above this comment would _definitely_ be clearer as a standard test case harness and I'm fairly even the tests below would be?

(i.e. I think the reftests framework is excellent for full-blown tests, but these feel more like unit tests and the framework's overkill for that?)

Reply:
I think it's fine to have in the reftest since it's something that is usable and most used from the command line tool